yt_dlp/extractor/patreon.py

```diff
@@ -7,17 +8,44 @@
 from ..utils import (
     clean_html,
     determine_ext,
+    ExtractorError,
     int_or_none,
     KNOWN_EXTENSIONS,
     mimetype2ext,
     parse_iso8601,
     str_or_none,
+    traverse_obj,
     try_get,
     url_or_none,
 )
 
 
-class PatreonIE(InfoExtractor):
+class PatreonBaseIE(InfoExtractor):
+    # XXX: should user-supplied user agent should override request user-agents?
+    USER_AGENT = 'Patreon/7.6.28'
+
+    def _call_api(self, ep, item_id, query=None, headers=None, fatal=True, note=None):
+        if headers is None:
+            headers = {}
+        if 'User-Agent' not in headers:
+            headers['User-Agent'] = self.USER_AGENT
+
+        try:
+            return self._download_json(
+                f'https://www.patreon.com/api/{ep}',
+                item_id, note='Downloading API JSON' if not note else note,
+                query=query, fatal=fatal, headers=headers)
+        except ExtractorError as e:
+            if isinstance(e.cause, HTTPError):
+                if e.cause.code == 403:  # TODO: catch and test for other error codes too
+                    err_json = self._parse_json(self._webpage_read_content(e.cause, None, item_id), item_id, fatal=False)
+                    err_message = traverse_obj(err_json, ('errors', ..., 'detail'), get_all=False)
+                    if err_message:
+                        raise ExtractorError(f'Patreon said: {err_message}', expected=True)
+            raise
```

Comment:
You can de-indent this a level with an early raise

Reply:
not quite sure what you mean? I want it to re-raise it if we don't end up raising our own

Reply:
I just meant:
```suggestion
            if not isinstance(e.cause, HTTPError) or e.cause.code != 403:
                raise
            err_json = self._parse_json(self._webpage_read_content(e.cause, None, item_id), item_id, fatal=False)
            err_message = traverse_obj(err_json, ('errors', ..., 'detail'), get_all=False)
            if err_message:
                raise ExtractorError(f'Patreon said: {err_message}', expected=True)
            raise
```

Though it's really only a personal preference, not a needed change